arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/flow.kt

```diff
@@ -0,0 +1,57 @@
+@file:JvmMultifileClass
+@file:JvmName("FlowExtensions")
+
+package arrow.fx.coroutines
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.retryWhen
+
+/**
+ * Retries collection of the given flow when an exception occurs in the upstream flow based on a decision by the [schedule].
+ * This operator is *transparent* to exceptions that occur in downstream flow and does not retry on exceptions that are thrown
+ * to cancel the flow.
+ *
+ * @see [Schedule] for how to build a schedule.
+ *
+ * ```kotlin:ank:playground
+ * import kotlinx.coroutines.flow.*
+ * import arrow.fx.coroutines.*
+ * suspend fun main(): Unit {
+ *   var counter = 0
+ *   val flow = flow {
+ *    emit(a)
+ *    if (++counter <= 5) throw RuntimeException("Bang!")
+ *   }
+ *   //sampleStart
+ *  val sum = flow.retry(Schedule.recurs(5))
+ *    .reduce { acc, int -> acc + int }
+ *   //sampleEnd
+ *   println(result)
+ * }
+ * ```
+ *
+ * @param schedule - the [Schedule] used for retrying the collection of the flow
+ */
+fun <A, B> Flow<A>.retry(schedule: Schedule<Throwable, B>): Flow<A> = flow {
+  (schedule as Schedule.ScheduleImpl<Any?, Throwable, B>)
+  var dec: Schedule.Decision<Any?, B>
+  var state: Any? = schedule.initialState()
+
+  val retryWhen = retryWhen { cause, _ ->
+    dec = schedule.update(cause, state)
```

Comment:
Shouldn't we check for a possible cancellation exception here and rethrow?

Reply:
That should be handled by `retryWhen` 

https://github.com/Kotlin/kotlinx.coroutines/blob/22087ef122b5ac677fd14941a18b6aefc57393d2/kotlinx-coroutines-core/common/src/flow/operators/Errors.kt#L207

Reply:
You're right, thank you!